uefi-mkconfig

```diff
@@ -13,152 +13,112 @@ ewarn() {
 
-		# Add new entry
-		if [[ -z $backup_efi ]]; then
-			efibootmgr -q --create -b "$bootnum" --disk /dev/"$partition" --label "$entry_label" --loader "${efi_file_path//\//\\}"\
-			-u "$adding_kernel_commands" || die "Failed to add UEFI entry for \"$efi_file_path\""
-		else
-			efibootmgr -q --create-only -b "0100" --disk /dev/"$partition" --label "$entry_label" --loader "${efi_file_path//\//\\}"\
+	# Add new entry
+	if [[ -n $backup_efi ]]; then
+		# When creating backup entry, don't add it to the bootorder
+		efibootmgr -q --create-only -b "$bootnum" --disk /dev/"$partition" --label "$entry_label" --loader "${efi_file_path//\//\\}"\
+			-u "$adding_kernel_commands" || die "Failed to add BACKUP UEFI entry for \"$efi_file_path\""
+	else
+		efibootmgr -q --create -b "$bootnum" --disk /dev/"$partition" --label "$entry_label" --loader "${efi_file_path//\//\\}"\
 			-u "$adding_kernel_commands" || die "Failed to add UEFI entry for \"$efi_file_path\""
-		fi
-	done
+	fi
 )
 
-backup_entry () {
-	backup_efi=
-
-	# Check if backup entry exist at ID 200
-	if [[ "" == "$(efibootmgr -u | grep "Boot0100")" ]]; then
-		for efi_file in $partition_efis; do	
-			# Check if file marking backup kernel image exists
-			if [[ -f "${efi_file}.uefibackup" ]]; then 
-				backup_efi="$efi_file"
-			fi
-		done
+wipe_entries () {
+	local initial_uefi_umc_state
 
-		# Create backup entry from the most recent kernel image with backup file mark
-		if [[ -n $backup_efi ]]; then 
-			partition_efis="$backup_efi"
-			add_uefi_entries
-			backup_efi=
-		else
-			ewarn "No backup kernel image designated!"
-		fi
-	else
-		einfo "Backup UEFI entry found at 0100"
-	fi
+	# Get list of all non backup entries made by uefi-mkconfig
+	initial_uefi_umc_state="$(printf %s "$initial_uefi_state" | grep " UMC ")"
+	
+	local IFS=$'\n'
+	for delete_entry in $initial_uefi_umc_state; do
+		efibootmgr -q -B -b "${delete_entry:4:4}"
+	done
 }
 
 main () {
 	efi_parttype="c12a7328-f81f-11d2-ba4b-00a0c93ec93b"
-	mounted_efi_partitions=$(lsblk -lo NAME,MOUNTPOINTS,PARTTYPE | grep  "$efi_parttype" | grep "/" | cut -d' ' -f1 | sort -r)
-	proc_kernel_commands=
+	mounted_efi_partitions=$(lsblk -lno MOUNTPOINTS,PARTTYPE | grep "$efi_parttype")
 	valid_kernel_commands=
 	kernel_prefixes="vmlinuz vmlinux- kernel- bzImage zImage"
 	
 	einfo "Running uefi-mkconfig..."
 
-	[[ ${EUID} -eq 0 ]] || die "Please run uefi-mkconfig as root!"
+	if [[ ${EUID} -ne 0 ]]; then
+		die "Please run uefi-mkconfig as root!"
 
-	[[ -n ${mounted_efi_partitions} ]] || die "No mounted efi partitions!"
+	elif [[ -z $(lsblk -o PARTTYPE | grep -v -e "PARTTYPE" -e "^$") ]]; then 
+		die "lsblk can't see PARTTYPE! Please refer to Troubleshooting section in README or open an issue"
 
-	[[ -n "$(command -v efibootmgr)" ]] || die "efibootmgr dependency not found!" 
+	elif [[ -z ${mounted_efi_partitions} ]]; then
+		die "No mounted efi partitions!"
+
+	elif [[ ! -x "$(which efibootmgr)" ]]; then
+		die "efibootmgr dependency not found!"
+
```

Comment:
Don't use which (https://bugs.gentoo.org/646588). `command -v` is the built-in POSIX way to do this.